### PR TITLE
[BEAM-2310] support new data type: DECIMAL

### PR DESCRIPTION
**Pull request by xumingming**
1. support new data type: DECIMAL
2. support encoding/decoding for TIME
3. add tests all the supported data types.

R: @XuMingmin

### CODE REVIEW

**Comment by jbonofre** [comment]
R: @jbonofre

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlRowCoder.java [hunk 1]
```diff
@@ -81,22 +85,29 @@ public void encode(BeamSQLRow value, OutputStream outStream,
...
+          longCoder.encode(value.getDate(idx).getTime(), outStream, context.nested());
           break;
 
         default:
           throw new UnsupportedDataTypeException(value.getDataType().getFieldsType().get(idx));
       }
     }
 
-    instantCoder.encode(value.getWindowStart(), outStream, context.nested());
+    instantCoder.encode(value.getWindowStart(), outStream, context);
```
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlRowCoder.java]: would keep as `context.nested()`, and only has the last field in coder/decoder as `context`, 
Same for method `decode`.

Comment (xumingming) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlRowCoder.java]: Is there any criteria for this reason? I don't quite understand the `context/context.nested()` thing, my current strategy is just to keep consistent in `encode/decode`.

**Comment by xumingming** [comment]
@jbonofre can you take a look at this one?

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlRowCoder.java (hunk 1 above)
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlRowCoder.java]: I would follow the pattern to have only the last field as `context`, and `context().nested()` with others.

**Comment by jbonofre** [comment]
Merged on `DSL_SQL` branch, you can close this Pull Request. Thanks !

**Comment by xumingming** [comment]
Thanks!

**Pull request closed without merging**